cli/displayAdsRecorder.js

```diff
@@ -46,83 +46,91 @@ const findAdsInDirectory = require("../src/util/findAdsInDirectory");
-    name: "output",
-    message: "Please select output(s)",
-    validate: (answers) => answers.length > 0,
-    choices: [
-      { name: "mp4", value: "mp4", checked: false },
-      { name: "gif (animated)", value: "gif", checked: false },
-      { name: "jpg (last frame)", value: "jpg", checked: false },
-    ],
-  });
-
-  configQuestions.push({
-    type: "list",
-    name: "gifLoopOptions",
-    when: output.includes("gif"),
-    message:
-      "You selected .gif as additional output. Would you like it run once or loop?",
-    choices: [
-      { name: "Run once", value: "-1" },
-      { name: "Loop", value: "0" },
-    ],
-    default: 0,
-  });
-
-  configQuestions.push({
-    type: "list",
-    name: "fps",
-    when: output.includes("mp4") || output.includes("gif"),
-    message: "Please select fps to record at",
-    choices: [{ name: 15 }, { name: 30 }, { name: 60 }],
-    default: 1,
-  });
-
-  const { jpgMaxFileSize } = options.backup
-  ? { jpgMaxFileSize: options.backup }
-  : await inquirer.prompt({
-    type: "input",
-    name: "jpgMaxFileSize",
-    when: output.includes("jpg"),
-    message: "Please select max KB filesize for backup image",
-    default: 40,
-  });
-
-  const adSelection = await inquirer.prompt(configQuestions);
-
-  adSelection.location = options.all
-  ? location
-  : (location.indexOf("all") > -1)
-    ? allAds
-    : location;
-
-  adSelection.output = output
-  adSelection.jpgMaxFileSize = jpgMaxFileSize
+      type: "checkbox",
+      name: "output",
+      message: "Please select output(s)",
+      validate: (answers) => answers.length > 0,
+      choices: outputChoices
+    });
+
+  const { gifLoopOptions } = output.includes("gif")
+  ? options.gif != true
+    ? { gifLoopOptions: options.gif }
+    : await inquirer.prompt({
+        type: "list",
+        name: "gifLoopOptions",
+        message:
+          "You selected .gif as additional output. Would you like it run once or loop?",
+        choices: [
+          { name: "Run once", value: "-1" },
+          { name: "Loop", value: "0" },
+        ],
+        default: 0,
+      })
+  : { gifLoopOptions: undefined }
+
+  const { fps } = (output.includes("mp4") || output.includes("gif"))
+  ? options.fps
+    ? options
+    : await inquirer.prompt({
+        type: "list",
+        name: "fps",
+        message: "Please select fps to record at",
+        choices: [{ name: 15 }, { name: 30 }, { name: 60 }],
+        default: 1,
+      })
+  : { fps: undefined }
+
+  const { jpgMaxFileSize } = output.includes("jpg")
+  ? options.jpg == false // if jpg is selected via answers - ask for KB, otherwise use specified or default value
+    ? await inquirer.prompt({
+        type: "input",
+        name: "jpgMaxFileSize",
+        message: "Please select max KB filesize for backup image",
+        default: 40,
+      })
+    : options.jpg != true
+      ? { jpgMaxFileSize: options.jpg }
+      : { jpgMaxFileSize: 40 }
+  : { jpgMaxFileSize: undefined }
```

Comment:
Instead of storing all values you are putting to objects to undefined by default, I think it is better to push all used vars into an array like previously was done with const configQuestions = [];?

Reply:
let's look into this together later on (since you are ooo tomorrow)

Reply:
@mm-vasyl , let me try to explain. Now you have "undefined" as an option for  a couple of variables such as jpgMaxFileSize. What happens when for example jpgMaxFileSize is undefined but IS used in index.js? In the `function recordBackup(adLocation), maxSizeBytes: adSelection.jpgMaxFileSize * 1024,` isn't that then a problem? Shouldn't the variables be only pushed when filled in/available? Or fall back on a default value at least? :)  

The previous code had this kind of setup where there was an array pushing only the values that were defined. Let me know if you understand or if I am overseeing something :)